relayer/relayer.go

```diff
@@ -394,6 +467,43 @@ func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, storeProcessedH
 
 // Returns whether destinationBlockchainID is a supported destination.
 func (r *Relayer) CheckSupportedDestination(destinationBlockchainID ids.ID) bool {
-	supportedDsts := r.sourceBlockchain.GetSupportedDestinations()
-	return supportedDsts.Contains(destinationBlockchainID)
+	for _, destination := range r.sourceBlockchain.SupportedDestinations {
+		if destination.GetBlockchainID() == destinationBlockchainID {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Relayer) CheckAllowedOriginSenderAddress(addr common.Address) bool {
+	allowedAddresses := r.globalConfig.GetSourceBlockchainAllowedAddresses(r.sourceBlockchain.GetBlockchainID())
+	if bytes.Equal(allowedAddresses[0].Bytes(), common.Address{}.Bytes()) {
+		return true
+	}
+	for _, allowedAddress := range allowedAddresses {
```

Comment:
Seems like it may make sense to make `allowedAddresses` a `Set` rather than having to iterate it each time we see a message here.

Reply:
This method has been removed.